op-interop-filter/filter/service.go

```diff
@@ -0,0 +1,242 @@
+		}
+		if err := cfg.Check(); err != nil {
+			return nil, fmt.Errorf("invalid config: %w", err)
+		}
+
+		l := oplog.NewLogger(oplog.AppOut(cliCtx), cfg.LogConfig)
+		oplog.SetGlobalLogHandler(l.Handler())
+		opservice.ValidateEnvVars(flags.EnvVarPrefix, flags.Flags, l)
+
+		l.Info("Initializing op-interop-filter", "version", version)
+		return NewService(cliCtx.Context, cfg, l)
+	}
+}
+
+// NewService creates a new Service instance
+func NewService(ctx context.Context, cfg *Config, logger log.Logger) (*Service, error) {
+	s := &Service{
+		log:     logger,
+		version: cfg.Version,
+	}
+	if err := s.init(ctx, cfg); err != nil {
+		return nil, errors.Join(err, s.Stop(ctx))
+	}
+	return s, nil
+}
+
+func (s *Service) init(ctx context.Context, cfg *Config) error {
+	s.initMetrics(cfg)
+
+	if err := s.initPProf(cfg); err != nil {
+		return fmt.Errorf("failed to init pprof: %w", err)
+	}
+	if err := s.initMetricsServer(cfg); err != nil {
+		return fmt.Errorf("failed to init metrics server: %w", err)
+	}
+	if err := s.initBackend(ctx, cfg); err != nil {
+		return fmt.Errorf("failed to init backend: %w", err)
+	}
+	if err := s.initRPCServer(cfg); err != nil {
+		return fmt.Errorf("failed to init RPC server: %w", err)
+	}
+	return nil
+}
+
+func (s *Service) initMetrics(cfg *Config) {
+	if cfg.MetricsConfig.Enabled {
+		s.metrics = metrics.NewMetrics("default")
+		s.metrics.RecordInfo(s.version)
+	} else {
+		s.metrics = metrics.NoopMetrics
+	}
+}
+
+func (s *Service) initPProf(cfg *Config) error {
+	s.pprofService = oppprof.New(
+		cfg.PprofConfig.ListenEnabled,
+		cfg.PprofConfig.ListenAddr,
+		cfg.PprofConfig.ListenPort,
+		cfg.PprofConfig.ProfileType,
+		cfg.PprofConfig.ProfileDir,
+		cfg.PprofConfig.ProfileFilename,
+	)
+	if err := s.pprofService.Start(); err != nil {
+		return fmt.Errorf("failed to start pprof: %w", err)
+	}
+	return nil
+}
+
+func (s *Service) initMetricsServer(cfg *Config) error {
+	if !cfg.MetricsConfig.Enabled {
+		s.log.Info("Metrics disabled")
+		return nil
+	}
+	m, ok := s.metrics.(opmetrics.RegistryMetricer)
+	if !ok {
+		return fmt.Errorf("metrics do not expose registry")
+	}
+	metricsSrv, err := opmetrics.StartServer(m.Registry(), cfg.MetricsConfig.ListenAddr, cfg.MetricsConfig.ListenPort)
+	if err != nil {
+		return fmt.Errorf("failed to start metrics server: %w", err)
+	}
+	s.log.Info("Started metrics server", "addr", metricsSrv.Addr())
+	s.metricsSrv = metricsSrv
+	return nil
+}
+
+func (s *Service) initBackend(ctx context.Context, cfg *Config) error {
+	backend, err := NewBackend(ctx, s.log, s.metrics, cfg)
+	if err != nil {
+		return err
+	}
+	s.backend = backend
+	return nil
+}
+
+func (s *Service) initRPCServer(cfg *Config) error {
+	opts := []oprpc.Option{
+		oprpc.WithLogger(s.log),
+	}
+
+	// Load JWT secret if path is provided (generates new secret if file is empty)
+	if cfg.JWTSecretPath != "" {
+		secret, err := oprpc.ObtainJWTSecret(s.log, cfg.JWTSecretPath, true)
```

Comment:
Note: If `cfg.JWTSecretPath` is empty, there will be no auth. We may add the check that `cfg.JWTSecretPath` is not empty at `func NewConfig(ctx *cli.Context, version string)`